From 22c98354053baee3c9a93eb552d8767c2be60f7a Mon Sep 17 00:00:00 2001
From: Ben Pfaff <blp@nicira.com>
Date: Mon, 20 Jun 2011 14:14:27 -0700
Subject: [PATCH 18/19] ovsdb-doc: Add support for references to specific keys.

Now a specific key can be referenced with syntax like
<ref table="Bridge" column="other-config" key="datapath-id"/>.

Also fixes up an existing place that needed this feature already (and had
a typo, too).
---
 ovsdb/ovsdb-doc.in   |    2 ++
 vswitchd/vswitch.xml |    7 +++----
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/ovsdb/ovsdb-doc.in b/ovsdb/ovsdb-doc.in
index 5ba4e71..83643c3 100755
--- a/ovsdb/ovsdb-doc.in
+++ b/ovsdb/ovsdb-doc.in
@@ -52,6 +52,8 @@ def inlineXmlToNroff(node, font):
             s = r'\fB'
             if node.hasAttribute('column'):
                 s += node.attributes['column'].nodeValue
+                if node.hasAttribute('key'):
+                    s += ':' + node.attributes['key'].nodeValue
             elif node.hasAttribute('table'):
                 s += node.attributes['table'].nodeValue
             elif node.hasAttribute('group'):
diff --git a/vswitchd/vswitch.xml b/vswitchd/vswitch.xml
index ad1bffa..2b12e29 100644
--- a/vswitchd/vswitch.xml
+++ b/vswitchd/vswitch.xml
@@ -363,10 +363,9 @@
       </column>
 
       <column name="datapath_id">
-        Reports the OpenFlow datapath ID in use.  Exactly 16 hex
-        digits.  (Setting this column will have no useful effect.  Set
-        <ref column="other_config"/>:<code>other-config</code>
-        instead.)
+        Reports the OpenFlow datapath ID in use.  Exactly 16 hex digits.
+        (Setting this column has no useful effect.  Set <ref
+        column="other-config" key="datapath-id"/> instead.)
       </column>
     </group>
 
-- 
1.6.3.3

